libs/getdns/Makefile

```diff
@@ -55,6 +55,8 @@ CONFIGURE_ARGS += \
 		$(if $(CONFIG_GETDNS_ENABLE_IDN_LIBIDN2), , --without-libidn2 ) \
 		--with-ssl="$(STAGING_DIR)/usr" \
 
+CONFIGURE_VARS += LIBBSD_LIBS=-lc
+
```

Comment:
Please add a note in the preceding line explaining the hack.

By the way, in my opinion it's actually better if we make it explicit that the package depends on libbsd, becase that is indeed the case.  This way of code reuse is one benefit what shared library is for.  It's good for code size in the full picture. Otherwise getdns will use its own static implementation which will be a bloat when libbsd is actually on the target list.

Reply:
@yousong 
Is this a good choose?
```
diff --git a/libs/getdns/Makefile b/libs/getdns/Makefile
index 2578659..a5080f9 100644
--- a/libs/getdns/Makefile
+++ b/libs/getdns/Makefile
@@ -22,7 +22,8 @@ PKG_INSTALL:=1
 
 PKG_CONFIG_DEPENDS:= \
        CONFIG_GETDNS_ENABLE_STUB_ONLY \
-       CONFIG_GETDNS_ENABLE_IDN_LIBIDN2 
+       CONFIG_GETDNS_ENABLE_IDN_LIBIDN2 \
+       CONFIG_PACKAGE_libbsd
        
 include $(INCLUDE_DIR)/package.mk
 
@@ -36,7 +37,7 @@ define Package/getdns
        SECTION:=libs
        CATEGORY:=Libraries
        TITLE+= (library)
-       DEPENDS+= +libopenssl +!GETDNS_ENABLE_STUB_ONLY:libunbound +GETDNS_ENABLE_IDN_LIBIDN2:libidn2
+       DEPENDS+= +libopenssl +PACKAGE_libbsd:libbsd +!GETDNS_ENABLE_STUB_ONLY:libunbound +GETDNS_ENABLE_IDN_LIBIDN2:libidn2
        MENU:=1
 endef
 
@@ -55,7 +56,9 @@ CONFIGURE_ARGS += \
                $(if $(CONFIG_GETDNS_ENABLE_IDN_LIBIDN2), , --without-libidn2 ) \
                --with-ssl="$(STAGING_DIR)/usr" \
 
+ifeq ($(CONFIG_PACKAGE_libbsd),)
 CONFIGURE_VARS += LIBBSD_LIBS=-lc
+endif
 
 define Build/InstallDev
        $(INSTALL_DIR) $(1)/usr/include/getdns/
```

Reply:
No.  The 1st thing is that I as a user will be surprised if selecting `libbsd` will affect build result of `getdns`.  The 2nd thing is that I will have to check the build code to see if `libbsd` will be recorded as a dependency in the final .ipk file (if not then `getdns` will not run on target board if `CONFIG_PACKAGE_libbsd=m`).  Just select it with `+libbsd` should do.